memory/memory/memory.py

```diff
@@ -0,0 +1,100 @@
+import aioredis
+import asyncio
+import base64
+import concurrent
+import json
+import logging
+import os
+import uvloop
+from aiohttp import web
+import kubernetes_asyncio as kube
+
+from hailtop.config import get_deploy_config
+from hailtop.google_storage import GCS
+from hailtop.hail_logging import AccessLogger
+from hailtop.tls import get_in_cluster_server_ssl_context
+from hailtop.utils import AsyncWorkerPool, retry_transient_errors
+from gear import setup_aiohttp_session, rest_authenticated_users_only
+
+uvloop.install()
+
+BATCH_PODS_NAMESPACE = os.environ['HAIL_BATCH_PODS_NAMESPACE']
+log = logging.getLogger('batch')
+routes = web.RouteTableDef()
+
+socket = '/tmp/redis.sock'
+
+@routes.get('/healthcheck')
+async def healthcheck(request):  # pylint: disable=unused-argument
+    return web.Response()
+
+@routes.post('/getfile')
+@rest_authenticated_users_only
+async def getfile(request, userdata):
```

Comment:
Few comments on the API:

 - I think it is more common to call these objects instead of files in this context.
 - APIs should be versioned.  Our API URL naming scheme is `/api/<version>/...`.  Work-in-progress APIs should be `v1alpha`.
 - This should be GET, not POST
 - GET requests should not have a semantically meaningful body.  That means the object should go into the URL or inside they query parameters.  For fixed objects, the URL seems more natural.  So it would be something like `GET /api/v1alpha/objects/<bucket>/<key>`.  See:

 > https://stackoverflow.com/questions/978061/http-get-with-request-body
 > Server semantics for GET, however, are restricted such that a body, if any, has no semantic meaning to the request.